msteams-platform/samples/virtual-assistant.md

```diff
@@ -0,0 +1,560 @@
+
+```
+> ! Automatically generated by [LUDown CLI](https://github.com/Microsoft/botbuilder-tools/tree/master/Ludown), Tue Mar 31 2020 17:30:32 GMT+0530 (India Standard Time)
+
+> ! Source LUIS JSON file: book-a-meeting.json
+
+> ! Source QnA TSV file: Not Specified
+
+> ! Source QnA Alterations file: Not Specified
+
+
+> # Intent definitions
+
+## BOOK ROOM
+- book a room
+- book room
+- please book a room
+- reserve a room
+- i want to book a room
+- i want to book a room please
+- get me a room please
+- get me a room
+
+
+## MANAGE FAVORITES
+- manage favorites
+- manage favorite
+- please manage my favorite rooms
+- manage my favorite rooms please
+- manage my favorite rooms
+- i want to manage my favorite rooms
+
+## None
+
+
+> # Entity definitions
+
+
+> # PREBUILT Entity definitions
+
+
+> # Phrase list definitions
+
+
+> # List entities
+
+> # RegEx entities
+```
+
+With this approach, any command issues by a user to Virtual Assistant related to `book room` or `manage favorites` can be identified as a command associated with Book-a-room bot and is forwarded to this skill.
+On the other hand, Book-a-room room bot needs to use LUIS model to understand these commands if they are not typed as is (for example: `I want to manage my favorite rooms`).
+
+### Multi-Language support
+
+For this example, we have only created a LUIS model with English culture. You can create LUIS models corresponding to other languages and add entry to `cognitivemodels.json`.
+
+```json
+{
+  "defaultLocale": "en-us",
+  "languageModels": {
+    "en-us": {
+      "luisAppId": "",
+      "luisApiKey": "",
+      "luisApiHost": ""
+    },
+    "<your_language_culture>": {
+      "luisAppId": "",
+      "luisApiKey": "",
+      "luisApiHost": ""
+    }
+  }
+}
+```
+
+In parallel, add corresponding `.lu` file in luisFolder path. Folder structure should be as follows:
+
+```bash
+| - luisFolder
+
+        | - en-us
+
+                | - book-a-meeting.lu
+
+        | - your_language_culture
+
+                | - book-a-meeting.lu
+```
+
+Update botskills command as follows to modify `languages` parameter:
+
+```json
+botskills connect --remoteManifest "<url to skill's manifest>" --luisFolder "<path to luisFolder>" --languages "en-us, your_language_culture" --cs
+```
+
+Virtual Assistant uses `SetLocaleMiddleware` to identify current locale and invoke corresponding dispatch model. (Bot framework activity has locale field which is used by this middleware.) We recommend to use the same for your skill as well. Book-a-room bot does not use this middleware and instead gets locale from Bot framework activity's [clientInfo entity](https://github.com/microsoft/botframework-sdk/blob/master/specs/botframework-activity/botframework-activity.md#clientinfo). This will be deprecated soon.
```

Comment:
Can we remove the part about deprecating "locale"? We haven't announced this for Teams, and let's not have this article be the source of that information :)

Reply:
done (Y)